programs/lib/flexpret_exceptions.c

```diff
@@ -4,16 +4,27 @@
 #include "flexpret_assert.h"
 
 #include <flexpret.h>
+#include <errno.h>
 
 typedef void (*isr_t)(void);
 
 static isr_t ext_int_handler;
 static isr_t ie_int_handler;
 static isr_t ee_int_handler;
 
+struct thread_ctx_t contexts[NUM_THREADS];
+
+#ifndef NDEBUG
+uint32_t __stack_chk_guard = STACK_GUARD_INITVAL;
+
+FP_TEST_OVERRIDE
```

Comment:
`FP_TEST_OVERRRIDE` is a macro that expands to __attribute__(("weak")) when the make variable TEST is set and to nothing otherwise. Basically, it makes it possible to override the "weak" function in a test environment. See `programs/tests/mt-c-tests/stack/stack.c` for an example on how this is used to test the stack protection.

The `FP_TEST_OVERRRIDE` macro is used to test newlib's `malloc_lock` functions as well. See `programs/tests/mt-c-tests/heap/heap.c`.